sphinx_automodapi/tests/test_automodsumm.py

```diff
@@ -201,3 +201,51 @@ def test_ams_cython(tmpdir, cython_testpackage):  # noqa
         result = f.read()
 
     assert result == ams_cython_expected
+
+
+# =============================================================================
+
+CLASS_RST = """
+:orphan:
+
+.. currentmodule:: {mod}
+
+.. autoclass:: {cls}
+""".strip()
+
+sorted_str = """
+Before
+
+.. automodsumm:: sphinx_automodapi.tests.example_module.classes
+    :sort:
+
+And After
+"""
+
+sorted_expected = """\
+.. currentmodule:: sphinx_automodapi.tests.example_module.classes
+
+.. autosummary::
+
+    Egg
+    Spam
+
+"""
+
+
+def test_sort(tmpdir):
```

Comment:
Since this is new test, no reason to use legacy `tmpdir`? Can you please switch to `tmp_path`? Thanks!

Reply:
Unfortunately this will require a larger refactor, because `run_sphinx_in_tmpdir` requires a tmpdir object! (the .strpath)

Reply:
Ah ok, I opened https://github.com/astropy/sphinx-automodapi/issues/188 . Thanks for the clarification!